src/gpio.rs

```diff
@@ -165,6 +165,77 @@ macro_rules! gpio {
                 }
             }
 
+            impl<MODE> ExtiPin for $PXx<Input<MODE>> {
+                /// Make corresponding EXTI line sensitive to this pin
+                fn make_interrupt_source(&mut self, syscfg: &mut SYSCFG) {
+                    let offset = 4 * ($extigpionr % 4);
+                    match self.i {
+                        0...3 => {
+                            syscfg.exticr1.modify(|r, w| unsafe {
+                                let mut exticr = r.bits();
+                                exticr = (exticr & !(0xf << offset)) | ($extigpionr << offset);
+                                w.bits(exticr)
+                            });
+                        },
+                        4...7 => {
+                            syscfg.exticr2.modify(|r, w| unsafe {
+                                let mut exticr = r.bits();
+                                exticr = (exticr & !(0xf << offset)) | ($extigpionr << offset);
+                                w.bits(exticr)
+                            });
+                        },
+                        8...11 => {
+                            syscfg.exticr3.modify(|r, w| unsafe {
+                                let mut exticr = r.bits();
+                                exticr = (exticr & !(0xf << offset)) | ($extigpionr << offset);
+                                w.bits(exticr)
+                            });
+                        },
+                        12...15 => {
+                            syscfg.exticr4.modify(|r, w| unsafe {
+                                let mut exticr = r.bits();
+                                exticr = (exticr & !(0xf << offset)) | ($extigpionr << offset);
+                                w.bits(exticr)
+                            });
+                        },
```

Comment:
This whole code repetition is slightly ugly, but I see no solution to avoid it.
At least you could drop the exticr intermediate variable to shorten the code to simply:
```
w.bits((r.bits() & !(0xf << offset)) | ($extigpionr << offset))
```